audio_offset_finder/audio_offset_finder.py

```diff
@@ -24,43 +24,122 @@
 
-def find_offset(file1, file2, fs=8000, trim=60*15, hop_length=128, win_length=256, nfft=512, plot=False):
+    Parameters
+    ----------
+    file1: string
+        A path to the reference file, in any format that FFMPEG can read
+    file2: string
+        A path to the comparison file, in any format that FFMPEG can read
+    fs: int
+        The sampling rate that the audio should be resampled to prior to MFCC calculation, in Hz
+    trim: int
+        The length to which input files will be truncated before processing, in seconds
+    hop_length: int
+        The number of samples (at the resampled rate "fs") to skip between each calculated MFCC frame
+    win_length: int
+        The length of the window function used to avoid transients adding spurious high frequencies to the MFCCs
+    nfft: int
+        The number of samples to use in the FFTs used to generate the MFCCs
+
+    Returns
+    -------
+    A dict containing the following:
+    time_offset (float): the most likely offset of file2 compared to file1, in seconds.  A positive value indicates that
+                         file2 starts after file1
+    frame_offset (int): the offset of file2 compared to file1, measured in MFCC frames
+    standard_score (float): the standard score of the highest correlation coefficient in the cross-correlation curve
+    correlation (numpy int array): the 1D array of correlation coefficients calculated for the two input files
+    time_scale: the scalar factor that is multiplied to frame offsets to convert them to time offsets
+    earliest_frame_offset (int): the earliest offset searched for a correlation.  Always negative.
+    latest_frame_offset (int): the latest offset searched for a correlation.  Always positive.
+
+    Throws
+    ------
+    InsufficientAudioException if the audio supplied is too short to analyse.
+    """
     tmp1 = convert_and_trim(file1, fs, trim)
     tmp2 = convert_and_trim(file2, fs, trim)
-    # Removing warnings because of 18 bits block size
-    # outputted by ffmpeg
-    # https://trac.ffmpeg.org/ticket/1843
-    warnings.simplefilter("ignore", wavfile.WavFileWarning)
-    a1 = wavfile.read(tmp1, mmap=True)[1] / (2.0 ** 15)
-    a2 = wavfile.read(tmp2, mmap=True)[1] / (2.0 ** 15)
-    # We truncate zeroes off the beginning of each signals
-    # (only seems to happen in ffmpeg, not in sox)
-    a1 = ensure_non_zero(a1)
-    a2 = ensure_non_zero(a2)
+    a1 = wavfile.read(tmp1, mmap=True)[1] / (2.0**15)
+    a2 = wavfile.read(tmp2, mmap=True)[1] / (2.0**15)
+    offset_dict = find_offset_between_buffers(a1, a2, fs, hop_length, win_length, nfft)
+    os.remove(tmp1)
+    os.remove(tmp2)
+    return offset_dict
+
+
+def find_offset_between_buffers(buffer1, buffer2, fs, hop_length=128, win_length=256, nfft=512, max_frames=2000):
+    """Find the offset time offset between two audio files.
+
+    This function takes in two numpy arrays (assumed to be PCM audio) and compares them using cross-correlation of
```

Comment:
Does it matter if the buffers don't contain values in the range 0 to 1? (Looking at `find_offset_between_files`, the samples are converted from 16-bit integer to 0 to 1 range)

Reply:
Integers in python 3 have unlimited precision and floats are unaffected by a constant scaling factor, so I don't think this is a limitation.  Because we only care about the relative value of the results, the buffers can even be different ranges to each other and the results will be unchanged.

I have experimentally removed the sample range conversion (not checked in) and the tests still ran correctly.

Reply:
My only concern was whether comparing a buffer with a given number range with another buffer with a different range would affect the output, and so there may be a need to document the expected number ranges.

Reply:
Nope, comparing two buffers with different number ranges should be A-OK.